SyllabusPlusSchedulerService/SchedulerService.cs

```diff
@@ -125,27 +147,62 @@ private void ScheduleRecordings()
                                             this.configSettings.PanoptoSite,
                                             this.configSettings.PanoptoUserName,
                                             this.configSettings.PanoptoPassword))
+                                    using (SessionManagementWrapper sessionManagementWrapper
+                                        = new SessionManagementWrapper(
+                                        this.configSettings.PanoptoSite,
+                                        this.configSettings.PanoptoUserName,
+                                        this.configSettings.PanoptoPassword))
                                     {
                                         // Schedule session id will determine if need to create or update/delete the corresponding schedule
                                         if (schedule.ScheduledSessionId == null || schedule.ScheduledSessionId == Guid.Empty)
                                         {
                                             log.Debug(schedule.SessionName + " is not associated with a session on the Panopto database. Attempting to schedule.");
-                                            result = remoteRecorderManagementWrapper.ScheduleRecording(schedule);
+                                            result = VerifyFolderAndScheduleRecording(sessionManagementWrapper, schedule, remoteRecorderManagementWrapper);
                                         }
                                         else
                                         {
                                             log.Debug(schedule.SessionName + " is already associated with a session on the Panopto database. Attempting to update schedule.");
-                                            result = remoteRecorderManagementWrapper.UpdateRecordingTime(schedule);
+                                            Session scheduledSession = sessionManagementWrapper.GetSessionById((Guid)schedule.ScheduledSessionId);
```

Comment:
Yeah it could throw. This is in a try-catch so the error would be written out to the event log in the catch.